### PR TITLE
DOC-9913: 4.0 Migration Guides

### PR DESCRIPTION
**Pull request by osfameron**

### CODE REVIEW

File: modules/concept-docs/pages/durability-replication-failure-considerations.adoc [hunk 1]
```diff
@@ -17,7 +17,15 @@ include::{version-server}@sdk:shared:partial$durability-replication-failure-cons
...
 include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=syncrep3]
 
-include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=older]
+
+== Older Server Versions
+// include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=older]
+If a version of Couchbase Server lower than 6.5 is being used then the fallback is 'client verified' durability.
+
+WARNING: Client Verified durability is supported in xref:3.2@durability-replication-failure-considerations.adoc#older-server-versions[Python SDK 3.2] but not in 4.0.
+Legacy support will be available in a later 4.x release.
```
Comment (RichardSmedley) [inline modules/concept-docs/pages/durability-replication-failure-considerations.adoc:26]: Is that a definite commitment from engg?

**Review by RichardSmedley — APPROVED** [review]
:+1:

File: modules/concept-docs/pages/durability-replication-failure-considerations.adoc (hunk 1 above)
Comment (thejcfactor) [reply modules/concept-docs/pages/durability-replication-failure-considerations.adoc:26]: Should be:  https://issues.couchbase.com/browse/CXXCBC-53

File: modules/howtos/pages/managing-connections.adoc [hunk 2]
```diff
@@ -55,6 +55,10 @@ include::example$managing_connections.py[tag=customenv,indent=0]
 [#connection-strings]
 == Connection Strings
 
+WARNING: The backend implementation of connection strings parameters changed substantially in 4.0 and is not currently fully documented.
+This will be resolved in a future 4.x release.
+See xref:project-docs:migrating-sdk-code-to-3.n.adoc#sdk4-specifics[more details on migrating to 4.0].
+
 A Couchbase connection string is a comma-delimited list of IP addresses and/or hostnames, optionally followed by a list of parameters.
 
 The parameter list is just like the query component of a URI; name-value pairs have an equals sign (`=`) separating the name and value, with an ampersand (`&`) between each pair.
```
Comment (thejcfactor) [inline modules/howtos/pages/managing-connections.adoc:64]: Can we mention that while this is possible, it is preferred to pass in options via ClusterOptions()?

Comment (thejcfactor) [inline modules/howtos/pages/managing-connections.adoc:65]: Can we mention that while this is possible, it is preferred to pass in options via ClusterOptions()?

Comment (thejcfactor) [reply modules/howtos/pages/managing-connections.adoc:65]: Sorry -- not to hold this up.  LGTM!

**Review by thejcfactor — APPROVED** [review]

File: modules/concept-docs/pages/response-time-observability.adoc [hunk 3]
```diff
@@ -3,10 +3,15 @@
...
 
 include::project-docs:partial$attributes.adoc[]
 
 [abstract]
 {description}
 
-include::{version-server}@sdk:shared:partial$rto.adoc[tag=rto]
+// include::{version-server}@sdk:shared:partial$rto.adoc[tag=rto]
+WARNING: Response Time Observability is not supported in Python SDK 4.0.
+View xref:3.2@response-time-observability[the Node.js SDK 3.2 docs]
```
Comment (maria-robobug) [inline modules/concept-docs/pages/response-time-observability.adoc]:
```suggestion
View xref:3.2@response-time-observability[the Python SDK 3.2 docs]
```

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 4]
```diff
@@ -13,10 +13,18 @@ The Document class and structure has been completely removed from the API, and t
 Retry behaviour is more proactive, and lazy bootstrapping moves all error handling to a single place.
 Individual behaviour changes across services are explained here.
 
+The current PHP SDK 4.0 is also based on the xref:project-docs:compatibility.adoc#sdk3.3[SDK API 3.3 specification], and offers an entirely new backend (Couchbase++) with better support for new features like Distributed ACID Transactions.
```
Comment (maria-robobug) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]:
```suggestion
The current Python SDK 4.0 is also based on the xref:project-docs:compatibility.adoc#sdk3.3[SDK API 3.3 specification], and offers an entirely new backend (Couchbase++) with better support for new features like Distributed ACID Transactions.
```

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 5]
```diff
@@ -173,109 +179,103 @@ include::example$migrating.py[tag=certauth,indent=0]
...
 
-----
-include::example$migrating.py[tag=simpleget,indent=0]
-----
+// [source,python]
+// ----
+// include::example$migrating.py[tag=simpleget,indent=0]
+// ----
 ////
+
```
Comment (maria-robobug) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc:217]: Should this be uncommented?

**Review by maria-robobug — APPROVED** [review]
👍 Just some minor comments, LGTM.

File: modules/howtos/pages/managing-connections.adoc (hunk 2 above)
Comment (osfameron) [reply modules/howtos/pages/managing-connections.adoc:65]: Thanks, it's a good call, will add a brief note

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc (hunk 5 above)
Comment (osfameron) [reply modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc:217]: I don't think so - looks like it was previously block-commented, so these are just making that explicit.

**Pull request merged**